src/rdkafka_metadata.c

```diff
@@ -735,22 +774,13 @@ rd_kafka_parse_Metadata(rd_kafka_broker_t *rkb,
                         continue;
                 }
 
-                /* Sort partitions by partition id */
-                qsort(md->topics[i].partitions, md->topics[i].partition_cnt,
-                      sizeof(*md->topics[i].partitions),
-                      rd_kafka_metadata_partition_id_cmp);
-                qsort(mdi->topics[i].partitions, md->topics[i].partition_cnt,
-                      sizeof(*mdi->topics[i].partitions),
-                      rd_kafka_metadata_partition_internal_cmp);
-
                 if (compute_racks)
```

Comment:
I have done this change but I think this should be part of the first topic loop itself as `tbuf` is getting populated in that loop only. @milindl - Can you confirm this part once?

Reply:
@pranavrth It seems it works in both places because it doesn't use `rd_kafka_broker_find_by_nodeid` that was giving problems, putting it there means we do it also for blacklisted topics. Fair, it's not using it but has the same guarantees that other topics in the metadata struct.

Reply:
Looks good now to me, with blacklisted topics being ignored

Reply:
If we chose previous way for performance we should also avoid sorting in the first loop. I see blacklisted topics can be much because they're regular expressions.

Reply:
We need to sort this one in the first loop

```              qsort(md->topics[i].partitions, md->topics[i].partition_cnt,
                      sizeof(*md->topics[i].partitions),
                      rd_kafka_metadata_partition_id_cmp);```
                      
                      Since it's being returned to user. The internal one can be sorted anywhere.

Reply:
But I think just for our own sanity we should keep the internal and non-internal topics consistent

Reply:
Looking at https://github.com/confluentinc/librdkafka/issues/1834 - it seems like topic blacklist already causes some secondary effects like the topic not being assigned to other clients if the leader has them in its blacklist.  So computation of racks being below the blacklist if-clause should be okay. 
But it's weird, that we are returning the blacklisted topics in metadata pointer even though we say that "Topic blacklist, a comma-separated list of regular expressions for matching topic names that should be ignored in broker metadata information as if the topics did not exist. "

Reply:
On the whole, I think moving both the sorts to just before the compute racks if clause is the best way, preserves how things were for pre-2.2.0 and also has no performance impact. What do you think, @emasab , @pranavrth ?

Reply:
You mean everything starting from the blacklist with continue in the second loop? It's ok for me

Reply:
Yes, that's exactly what I meant, thansk

Reply:
I have updated the PR. Moved sorting in second loop as well.